examples/cluster/plot_kmeans_digits.py

```diff
@@ -23,72 +22,163 @@
+    """Benchmark to evaluate the KMeans initialization methods.
+
+    Parameters
+    ----------
+    kmeans : KMeans instance
+        A :class:`~sklearn.cluster.KMeans` instance with the initialization
+        already set.
+    name : str
+        Name given to the strategy. It will be used to show the results in a
+        table.
+    data : ndarray of shape (n_samples, n_features)
+        The data to cluster.
+    labels : ndarray of shape (n_samples,)
+        The labels used to compute the clustering metrics which requires some
+        supervision.
+    """
+    t0 = time()
+    estimator = make_pipeline(StandardScaler(), kmeans).fit(data)
+    fit_time = time() - t0
+    results = [name, fit_time, estimator[-1].inertia_]
+
+    # Define the metrics which require only the true labels and estimator
+    # labels
+    clustering_metrics = [
+        metrics.homogeneity_score,
+        metrics.completeness_score,
+        metrics.v_measure_score,
+        metrics.adjusted_rand_score,
+        metrics.adjusted_mutual_info_score,
+    ]
+    results += [m(labels, estimator[-1].labels_) for m in clustering_metrics]
+
+    # The silhouette score requires the full dataset
+    results += [
+        metrics.silhouette_score(
+            data,
+            estimator[-1].labels_,
+            metric="euclidean",
+            sample_size=300,
+        )
+    ]
+
+    # Show the results
+    formatter_result = (
+        "{:9s}\t{:.3f}s\t{:.0f}\t{:.3f}\t{:.3f}"
+        "\t{:.3f}\t{:.3f}\t{:.3f}\t{:.3f}"
+    )
+    print(formatter_result.format(*results))
+
+
+# %%
+# Run the benchmark
+# -----------------
+#
+# We will compare three approaches:
+#
+# * an initialization using `kmeans++`. This method is stochastic and we will
+#   run the initialization 4 times;
+# * a random initialization. This method is stochastic as well and we will run
+#   the initialization 4 times;
+# * an initialization based on a :class:`~sklearn.decomposition.PCA`
+#   projection. Indeed, we will use the components of the
+#   :class:`~sklearn.decomposition.PCA` to initialize KMeans. This method is
+#   deterministic and a single initialization suffice.
 from sklearn.cluster import KMeans
-from sklearn.datasets import load_digits
 from sklearn.decomposition import PCA
-from sklearn.preprocessing import scale
-
-np.random.seed(42)
-
-X_digits, y_digits = load_digits(return_X_y=True)
-data = scale(X_digits)
-
-n_samples, n_features = data.shape
-n_digits = len(np.unique(y_digits))
-labels = y_digits
-
-sample_size = 300
-
-print("n_digits: %d, \t n_samples %d, \t n_features %d"
-      % (n_digits, n_samples, n_features))
-
 
 print(82 * '_')
 print('init\t\ttime\tinertia\thomo\tcompl\tv-meas\tARI\tAMI\tsilhouette')
 
+bench_k_means(
+    kmeans=KMeans(
+        init="k-means++",
+        n_clusters=n_digits,
+        n_init=4,
+        random_state=0,
+    ),
+    name="k-means++",
+    data=data,
+    labels=labels,
+)
+
+bench_k_means(
+    kmeans=KMeans(
```

Comment:
I see the advantage of black for library code, but for example code or any type of teaching material I find it harder to read. Having lines span horizontally feels more "human readable"

But this could be just me. I wonder what every one else thinks.

Reply:
I made the example more compact.